### PR TITLE
feat(react): add CollapsibleNavbarItem component

### PR DESCRIPTION
**Pull request by savindi7**
### Purpose

- Add `CollapsibleNavbarItem` component
- Modularize navbar item related components and simplify implementation (`NavbarItem` and `CollapsibleNavbarItem`)
- Improve styling in Navbar
- Improve interfaces

`CollapsibleNavbarItem`

Collapsed

Screenshot 2023-07-21 at 02 41 38

Screenshot 2023-07-21 at 02 41 47

Expanded

Screenshot 2023-07-21 at 02 43 08

Screenshot 2023-07-21 at 02 43 19

### Related Issues
- None

### Related PRs
- None

### Checklist
- [ ] UX/UI review done on the final implementation.
- [ ] Story provided. (Add screenshots)
- [ ] Manual test round performed and verified.
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Documentation provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: packages/react/src/components/CollapsibleNavbarItem/CollapsibleNavbarItem.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import {ArgsTable, Source, Story, Canvas, Meta} from '@storybook/addon-docs';
```
Comment (pavinduLakshan) [inline packages/react/src/components/CollapsibleNavbarItem/CollapsibleNavbarItem.stories.mdx:1]: License header is missing

File: packages/react/src/components/CollapsibleNavbarItem/CollapsibleNavbarItem.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (pavinduLakshan) [inline packages/react/src/components/CollapsibleNavbarItem/CollapsibleNavbarItem.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). 
```

File: packages/react/src/components/CollapsibleNavbarItem/__tests__/CollapsibleNavbarItem.test.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (pavinduLakshan) [inline packages/react/src/components/CollapsibleNavbarItem/__tests__/CollapsibleNavbarItem.test.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```

File: packages/react/src/components/CollapsibleNavbarItem/__tests__/__snapshots__/CollapsibleNavbarItem.test.tsx.snap [hunk 4]
```diff
@@ -0,0 +1,58 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (pavinduLakshan) [inline packages/react/src/components/CollapsibleNavbarItem/__tests__/__snapshots__/CollapsibleNavbarItem.test.tsx.snap:1]: We should add license header to snapshot files I guess. (Not manually but through some automated process maybe)

File: packages/react/src/components/CollapsibleNavbarItem/collapsible-navbar-item.scss [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+.oxygen-collapsible-navbar-item {
+  flex-shrink: 0;
+  white-space: nowrap;
+  box-sizing: border-box;
+  border-radius: var(--oxygen-shape-borderRadius);
+  transition: background-color 1s ease-out;
+  margin-top: 5px;
+  margin-bottom: 5px;
+
+  .oxygen-list {
```
Comment (pavinduLakshan) [inline packages/react/src/components/CollapsibleNavbarItem/collapsible-navbar-item.scss:38]: Seems like style blocks are differently indented from here onwards. Is that intentional? if it's not, let's fix indentation :)

File: packages/react/src/components/CollapsibleNavbarItem/index.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (pavinduLakshan) [inline packages/react/src/components/CollapsibleNavbarItem/index.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). 
```

**Review by brionmario — APPROVED** [review]

File: packages/react/src/components/NavbarItem/NavbarItem.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (pavinduLakshan) [inline packages/react/src/components/NavbarItem/NavbarItem.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```

File: packages/react/src/components/NavbarItem/__tests__/NavbarItem.test.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (pavinduLakshan) [inline packages/react/src/components/NavbarItem/__tests__/NavbarItem.test.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```

File: packages/react/src/components/NavbarItem/index.tsx [hunk 9]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (pavinduLakshan) [inline packages/react/src/components/NavbarItem/index.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```

File: packages/react/src/components/CollapsibleNavbarItem/CollapsibleNavbarItem.tsx (hunk 2 above)
Comment (savindi7) [reply packages/react/src/components/CollapsibleNavbarItem/CollapsibleNavbarItem.tsx:2]: This needs to be fixed in a different repo.

File: packages/react/src/components/CollapsibleNavbarItem/collapsible-navbar-item.scss (hunk 5 above)
Comment (savindi7) [reply packages/react/src/components/CollapsibleNavbarItem/collapsible-navbar-item.scss:38]: This is the formatting style of Oxygen.

File: packages/react/src/components/CollapsibleNavbarItem/__tests__/__snapshots__/CollapsibleNavbarItem.test.tsx.snap (hunk 4 above)
Comment (savindi7) [reply packages/react/src/components/CollapsibleNavbarItem/__tests__/__snapshots__/CollapsibleNavbarItem.test.tsx.snap:1]: we can add an issue for this.

File: packages/react/src/components/CollapsibleNavbarItem/CollapsibleNavbarItem.stories.mdx (hunk 1 above)
Comment (savindi7) [reply packages/react/src/components/CollapsibleNavbarItem/CollapsibleNavbarItem.stories.mdx:1]: Mdx files were exempted from adding the license header. If this is required, we will have to create an issue and update all the files.

**Pull request merged**